### PR TITLE
Small fixes of MoveFiles

### PR DESCRIPTION
**Pull request by EmmanuelMess**
## Description

Multiple changes to AsyncTasks to use the new Task system with error handling.

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [ ] Done  
  

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java [hunk 1]
```diff
@@ -22,39 +22,50 @@
...
+      String path) {
+    this.applicationContext = context.getApplicationContext();
     this.input = input;
-    this.openMode = openMode;
     rootMode = root;
     isRegexEnabled = regex;
     isMatchesEnabled = matches;
+
+    this.file = new HybridFile(openMode, path);
+    file.generateMode(this.applicationContext);
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java:68]: Any reason to generate open mode again, when we already have it while calling.

Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java:68]: It was the first thing run in the on background. I left it as is.

Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java]: fixed

**Comment by VishalNehra** [comment]
please rebase

**Comment by EmmanuelMess** [comment]
Forgot I made this PR :sweat_smile:

**Review by VishalNehra — APPROVED** [review]

**Pull request merged**